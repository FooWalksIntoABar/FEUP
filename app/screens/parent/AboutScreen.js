import React, { Component } from 'react'
import { Body, Header, Icon, Left, Right, Title, Button, Content, Container, Text, H1, H2 } from 'native-base'
import PropTypes from 'prop-types'

export default class AboutScreen extends Component {
  render () {
    return (
      <Container>
        <Header>
          <Left>
            <Button transparent onPress={() => this.props.navigation.pop()}>
              <Icon name='arrow-back' />
            </Button>
          </Left>
          <Body>
            <Title>Sobre a equipa</Title>
          </Body>
          <Right />
        </Header>
        <Content style={{ marginHorizontal: 8, marginVertical: 8 }}>
          <H1>Sobre a equipa</H1>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            Esta aplicação foi feita possível pelo esforço conjunto da equipa de desenvolvimento e fantasticamente pacientes product owners.
          </Text>
          <H2>Product Owners</H2>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            • Carla Oliveira - Mestrado Integrado em Psicologia{'\n'}
            • Mariana Toste - Mestrado Integrado em Psicologia
          </Text>
          <H2>Equipa de Desenvolvimento</H2>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            • Afonso Pinto - @afonsobspinto{'\n'}
            • Francisco Maria - @francismaria{'\n'}
            • João Castro - @joaopscastro{'\n'}
            • Miguel Mano - @aquelemiguel{'\n'}
            • Paulo Correia - @pipas{'\n'}
            • Rui Oliveira - @ruimoliveira{'\n'}
            • Sérgio Salgado - @chaotixkilla
          </Text>
        </Content>
      </Container>
    )
  }
}

AboutScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
