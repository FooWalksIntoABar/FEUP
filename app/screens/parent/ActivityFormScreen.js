import React, { Component } from 'react'
import { Body, Header, Icon, Left, Label, Title, Button, Form, Content, Container, Item, Input, Right } from 'native-base'
import PropTypes from 'prop-types'

import { ItemPicker } from '../../components/Parent/ItemPicker'
import { ColorPicker } from '../../components/Parent/ColorPicker'
import { DurationPickers } from '../../components/Parent/DurationPickers'
import { ImagePickerButtons } from '../../components/Parent/ImagePickerButtons'
import { BottomButton } from '../../components/Parent/BottomButton'
import { availableColors } from '../../styles/Colors'

const defaultState = {
  title: 'Atividade de testes',
  color: '#0074D9',
  time: {
    goal: '3',
    min: '1.5',
    max: '6'
  },
  photo: undefined,
  image: undefined,
  routine: 'Manhã',
  createActivity: true
}

export default class ActivityFormScreen extends Component {
  constructor (props) {
    super(props)

    const activity = this.props.navigation.getParam('activity')

    if (activity) {
      this.state = {
        ...activity,
        time: {
          goal: activity.time.goal.toString(),
          min: activity.time.min.toString(),
          max: activity.time.max.toString()
        }
      }
    } else this.state = defaultState

    this.routines = ['Manhã', 'Noite']

    this.onColorChange = this.onColorChange.bind(this)
    this.onImageChange = this.onImageChange.bind(this)
    this.onPhotoChange = this.onPhotoChange.bind(this)
  }

  onColorChange = code => {
    this.setState({ color: code })
  }

  onDurationChange = bundle => {
    if (bundle.goal !== undefined) this.calculateTime(bundle.goal, this.state.time.min, this.state.time.max, !bundle.interpolate)
    if (bundle.min !== undefined) this.calculateTime(this.state.time.goal, bundle.min, this.state.time.max, !bundle.interpolate)
    if (bundle.max !== undefined) this.calculateTime(this.state.time.goal, this.state.time.min, bundle.max, !bundle.interpolate)
  }

  calculateTime = (goal, min, max, interpolate) => {
    const time = interpolate
      ? { goal, min: (goal * 0.25).toString(), max: (goal * 2).toString() }
      : { goal, min, max }

    this.setState({ time })
  }

  onPhotoChange = (uri) => {
    this.setState({ photo: { uri } })
  }

  onImageChange = (image) => {
    this.setState({ image: image })
  }

  onRoutineChange = (index) => {
    this.setState({ routine: this.routines[index] })
  }

  removeActivity = () => {
    console.log('Remove Activity')
  }

  createActivity = () => {
    console.log('Create Activity')
  }

  editActivity = () => {
    console.log('Edit Activity')
  }

  render () {
    return (
      <Container>
        <Header style={{ backgroundColor: this.state.color }} androidStatusBarColor={this.state.color}>
          <Left>
            <Button transparent onPress={() => this.props.navigation.pop()}>
              <Icon name='arrow-back' />
            </Button>
          </Left>
          <Body>
            <Title>{this.state.title}</Title>
          </Body>
          {!this.state.createActivity && <Right>
            <Button transparent onPress={this.removeActivity}>
              <Icon name='md-trash' />
            </Button>
          </Right>}
        </Header>
        <Content>
          <Form>
            <Item stackedLabel>
              <Label>Nome</Label>
              <Input value={this.state.title} onChangeText={text => this.setState({ title: text })} />
            </Item>
            <Item stackedLabel>
              <Label>Rotina</Label>
              <ItemPicker items={this.routines} selected={this.state.routine} onValueChange={this.onRoutineChange} />
            </Item>
            <Item stackedLabel>
              <Label>Cor</Label>
              <ColorPicker color={this.state.color} colors={availableColors} onColorChange={this.onColorChange} />
            </Item>
            <DurationPickers color={this.state.color} time={this.state.time} onDurationChange={this.onDurationChange} />
            <ImagePickerButtons color={this.state.color} onImageChange={this.onImageChange} onPhotoChange={this.onPhotoChange} photo={this.state.photo} image={this.state.image} />
            <BottomButton color={this.state.color} text={this.state.createActivity ? 'Criar Actividade' : 'Editar Actividade'} onPress={this.state.createActivity ? this.createActivity : this.editActivity} />
          </Form>
        </Content>
      </Container>
    )
  }
}

ActivityFormScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
