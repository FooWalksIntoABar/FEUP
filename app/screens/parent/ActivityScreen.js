import React from 'react'
import { Content, Spinner } from 'native-base'
import { InvalidateList } from '../../components/Parent/InvalidateList'
import { SelectChildPicker } from '../../components/Parent/SelectChildPicker'
import EnvVars from '../../constants/EnviromentVars'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export class ActivityScreen extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      children: [],
      activities: [],
      selectedChild: 0,
      loading: true
    }

    this.removeStars = this.removeStars.bind(this)
  }

  componentDidMount () {
    this.getChildren()
  }

  getChildren () {
    let url = `${EnvVars.apiUrl}routine_manager/children?userEmail=${this.props.loggedUserEmail}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.setState({ children: JSON.parse(responseJson.response) }, this.getActivities)
        } else {
          console.log('Ta mal')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  getActivities () {
    let url = `${EnvVars.apiUrl}routine_manager/history?id=${this.state.children[this.state.selectedChild].id}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          const activities = responseJson.response === null ? [] : JSON.parse(responseJson.response)
          this.setState({ activities, loading: false })
        } else {
          console.log('Ta mal')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  removeStars = (index) => {
    fetch(EnvVars.apiUrl + 'routine_manager/remove-stars/', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        id: this.state.children[this.state.selectedChild].id,
        reward: this.state.activities[index].reward,
        activityID: this.state.activities[index].id
      })
    }).then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.setState({ activities: this.state.activities.filter((_, i) => i !== index) })
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  onChildChanged = child => {
    this.setState({ selectedChild: child, loading: true }, this.getActivities)
  }

  render () {
    if (this.state.loading) return (<Content contentContainerStyle={{ flexGrow: 1, justifyContent: 'center' }}><Spinner /></Content>)
    else {
      return (
        <Content>
          <SelectChildPicker children={this.state.children} selected={this.state.selectedChild} onChildChanged={this.onChildChanged} />
          <InvalidateList activities={this.state.activities} onInvalidatePress={this.removeStars} />
        </Content>
      )
    }
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    loggedUserEmail: state.user.email
  })
)(ActivityScreen)

ActivityScreen.propTypes = {
  loggedUserEmail: PropTypes.string.isRequired
}
