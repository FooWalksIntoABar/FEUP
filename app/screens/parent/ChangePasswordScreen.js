import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import styles from '../../styles/ParentStyles/ChangePasswordScreen.style'

export default class ChangePasswordScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      title: 'Alterar Password'
    }
  };
  render () {
    return (
      <View>
        <View style={styles.generalLayout}>
          <View style={{ paddingTop: 10, paddingBottom: 20 }}>
            <Text>Insira a sua antiga password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Password Antiga'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.container}>
            <Text>Insira a sua nova password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Nova Password'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.container}>
            <Text>Repita a sua nova password.</Text>
            <TextInput
              secureTextEntry
              placeholder='Nova Password'
              style={styles.inputContainer}
            />
          </View>
          <View style={styles.confirmBtn}>
            <TouchableOpacity onPress={() => { console.log('changed password') }}>
              <Text>Alterar Password</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    )
  }
}
