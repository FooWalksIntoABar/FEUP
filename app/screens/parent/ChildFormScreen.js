import React, { Component } from 'react'
import { Body, Header, Icon, Left, Right, Label, Title, Button, Form, Content, Container, Item, Input } from 'native-base'
import { BackHandler } from 'react-native'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { PhotoPickerButton } from '../../components/Parent/PhotoPickerButton'
import { BottomButton } from '../../components/Parent/BottomButton'
import EnvVars from '../../constants/EnviromentVars'

export class ChildFormScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      title: 'Adicionar Criança',
      color: '#0074D9',
      name: '',
      photo: undefined,
      afterRegister: this.props.navigation.getParam('afterRegisterScreen', false)
    }
    this.handlePress = this.handlePress.bind(this)
  }

  removeBackButton () {
    return true // To be called by andrdid back button
  }

  componentDidMount () {
    if (this.state.afterRegister) BackHandler.addEventListener('hardwareBackPress', this.removeBackButton)
  }

  componentWillUnmount () {
    if (this.state.afterRegister) BackHandler.removeEventListener('hardwareBackPress', this.removeBackButton)
  }

  onPhotoChange = (uri) => {
    this.setState({ photo: { uri } })
  }

  handlePress () {
    if (this.state.name) {
      fetch(EnvVars.apiUrl + 'routine_manager/add-child/', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          userEmail: this.props.loggedUserEmail,
          name: this.state.name,
          gender: 'M', // TODO: this.props.gender Gender deve ser um F ou um M
          image: 'image' // TODO: nome da imagem
        })
      }).then((response) => response.json())
        .then((responseJson) => {
          if (responseJson.status === '200') {
            console.log('Criança Adicionada')
            if (this.state.afterRegister) this.props.navigation.replace('MainMenu')
            else this.props.navigation.popToTop()
          } else {
            console.log('Criança Não Adicionada')
          }
          return responseJson
        })
        .catch((error) => {
          console.error(error)
        })
    }
  }

  render () {
    console.log(this.state.name)
    return (
      <Container>
        <Header style={{ backgroundColor: this.state.color }} androidStatusBarColor={this.state.color}>
          <Left>
            {!this.state.afterRegister && <Button transparent onPress={() => this.props.navigation.pop()}>
              <Icon name='arrow-back' />
            </Button>}
          </Left>
          <Body>
            <Title>{this.state.title}</Title>
          </Body>
          <Right />
        </Header>
        <Content>
          <Form>
            <Item stackedLabel>
              <Label>Nome</Label>
              <Input value={this.state.name} onChangeText={text => this.setState({ name: text })} />
            </Item>
            <PhotoPickerButton color={this.state.color} onPhotoChange={this.onPhotoChange} photo={this.state.photo} />
            <BottomButton color={this.state.color} text={'Adicionar Criança'} onPress={this.handlePress} />
          </Form>
        </Content>
      </Container>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    loggedUserEmail: state.user.email
  })
)(ChildFormScreen)

ChildFormScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  loggedUserEmail: PropTypes.string.isRequired
}
