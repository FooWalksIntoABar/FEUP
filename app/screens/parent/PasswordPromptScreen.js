import React, { Component } from 'react'
import { View, Text, Alert } from 'react-native'
import { Container, Content, Form, Item, Input, Label, Button, Icon, Spinner } from 'native-base'
import styles from '../../styles/ParentStyles/RegisterScreen.style'
import PropTypes from 'prop-types'
import { oppositeColor } from '../../styles/Colors'

export default class PasswordPromptScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      storedPW: 'asdf',
      inputPW: '',
      passwordError: null,
      passwordHadInteraction: false,
      PWIsSecure: true,
      loading: false
    }
  }

  verifyPassword = () => {
    if (this.state.storedPW === this.state.inputPW) this.props.navigation.replace('ParentMainMenuScreen')
    else {
      Alert.alert('Palavra-passe incorreta!', 'Tente novamente...', [
        { text: 'Esqueci-me', onPress: () => console.log('Esqueceu-se lol burro') },
        { text: 'OK', style: 'cancel' }
      ],
      { cancelable: false })
    }
  }

  render () {
    if (this.state.loading) {
      return (
        <Container style={styles.registerContainter}>
          <Content contentContainerStyle={styles.contentContainter}>
            <Spinner color={oppositeColor} />
          </Content>
        </Container>
      )
    } else {
      return (
        <Container style={styles.registerContainter}>
          <Content contentContainerStyle={styles.contentContainter}>
            <Text style={styles.registerTitle}>Bloqueio parental</Text>
            <Text style={styles.loginText}>Introduza a sua palavra-passe abaixo para aceder Ã  zona de pai.</Text>
            <Form>
              <View style={{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }}>
                <Item floatingLabel error={this.state.passwordError} success={!this.state.passwordError && this.state.passwordHadInteraction} style={styles.inputContainer}>
                  <Label style={styles.labelText}>Palavra-passe</Label>
                  <Input className='password' onChangeText={inputPW => this.setState({ inputPW })} style={styles.labelText} secureTextEntry={this.state.PWIsSecure} />
                </Item>
                <Button style={{ position: 'absolute', right: -10, bottom: -5 }} transparent onPress={() => this.setState({ PWIsSecure: !this.state.PWIsSecure })}>
                  <Icon name='eye' style={{ color: 'white' }} />
                </Button>
              </View>
              <Button rounded block primary style={styles.submitButton} onPress={() => this.verifyPassword()}>
                <Text style={styles.buttonText}>Entrar</Text>
              </Button>
            </Form>
          </Content>
        </Container>
      )
    }
  }
}

PasswordPromptScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
