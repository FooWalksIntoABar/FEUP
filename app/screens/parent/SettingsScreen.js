import React from 'react'
import { connect } from 'react-redux'
import { Switch } from 'react-native'
import PropTypes from 'prop-types'
import { List, Content, ListItem, Text, Right, Body, Picker, Separator } from 'native-base'
import { toggleActivityProgressType, toggleActivityTimer, toggleRoutinePlayType, toggleActivityFeedback, changeFeedbackFrequency, togglePlaySounds } from '../../actions/settingsActions'
import EnvVars from '../../constants/EnviromentVars'

class SettingsScreen extends React.Component {
  componentWillUnmount () {
    fetch(EnvVars.apiUrl + 'routine_manager/settings/', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        userEmail: this.props.loggedUserEmail,
        activityProgressType: this.props.activityProgressType,
        activityShowTimer: this.props.activityShowTimer,
        activityFeedback: this.props.activityFeedback,
        feedbackFrequency: this.props.feedbackFrequency,
        routinePlayType: this.props.routinePlayType,
        playSounds: this.props.playSounds
      })
    }).then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          console.log('salvo')
        } else {
          console.log('oops')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  render () {
    return (
      <Content>
        <List>
          <Separator bordered>
            <Text>CRIANÇAS</Text>
          </Separator>
          <ListItem button icon onPress={() => this.props.navigation.navigate('ChildFormScreen')}>
            <Body><Text>Adicionar nova criança</Text></Body>
          </ListItem>
          <ListItem button icon onPress={() => this.props.navigation.navigate('RemoveChildScreen')}>
            <Body><Text>Remover criança</Text></Body>
          </ListItem>
          <Separator bordered>
            <Text>DEFINIÇÕES</Text>
          </Separator>
          <ListItem icon>
            <Body><Text>Visual de progresso</Text></Body>
            <Right>
              <Picker style={{ width: 125 }} selectedValue={this.props.activityProgressType} onValueChange={this.props.toggleActivityProgressType}>
                <Picker.Item label='Relógio' value='clock' />
                <Picker.Item label='Barra horizontal' value='bar' />
              </Picker>
            </Right>
          </ListItem>
          <ListItem icon>
            <Body><Text>Mostrar tempo restante</Text></Body>
            <Right><Switch value={this.props.activityShowTimer} onValueChange={this.props.toggleActivityTimer} /></Right>
          </ListItem>
          <ListItem icon>
            <Body><Text>Reproduzir sons</Text></Body>
            <Right><Switch value={this.props.playSounds} onValueChange={this.props.togglePlaySounds} /></Right>
          </ListItem>
          <ListItem icon>
            <Body><Text>Modo de reprodução de rotina</Text></Body>
            <Right>
              <Picker style={{ width: 125 }} selectedValue={this.props.routinePlayType} onValueChange={this.props.toggleRoutinePlayType}>
                <Picker.Item label='Escolha' value='choose' />
                <Picker.Item label='Automático' value='auto' />
              </Picker>
            </Right>
          </ListItem>
          <ListItem icon>
            <Body><Text>Tipo de feedback</Text></Body>
            <Right>
              <Picker style={{ width: 125 }} selectedValue={this.props.activityFeedback} onValueChange={this.props.toggleActivityFeedback}>
                <Picker.Item label='Vibração' value='vibration' />
                <Picker.Item label='Visual' value='visual' />
                <Picker.Item label='Sonoro' value='sound' enabled={this.props.playSounds} />
              </Picker>
            </Right>
          </ListItem>
          <ListItem icon>
            <Body><Text>Intensidade de feedback</Text></Body>
            <Right>
              <Picker style={{ width: 125 }} selectedValue={this.props.feedbackFrequency} onValueChange={this.props.changeFeedbackFrequency}>
                <Picker.Item label='Leve' value='slow' />
                <Picker.Item label='Moderado' value='normal' />
                <Picker.Item label='Intenso' value='fast' />
              </Picker>
            </Right>
          </ListItem>
        </List>
      </Content>
    )
  }
}

export default connect(
  state => ({
    activityProgressType: state.settings.activityProgressType,
    activityShowTimer: state.settings.activityShowTimer,
    activityFeedback: state.settings.activityFeedback,
    feedbackFrequency: state.settings.feedbackFrequency,
    routinePlayType: state.settings.routinePlayType,
    playSounds: state.settings.playSounds,
    loggedUserEmail: state.user.email
  }),
  dispatch => ({
    toggleActivityProgressType: () => dispatch(toggleActivityProgressType()),
    toggleActivityTimer: showTimer => dispatch(toggleActivityTimer(showTimer)),
    toggleActivityFeedback: () => dispatch(toggleActivityFeedback()),
    toggleRoutinePlayType: () => dispatch(toggleRoutinePlayType()),
    changeFeedbackFrequency: () => dispatch(changeFeedbackFrequency()),
    togglePlaySounds: () => dispatch(togglePlaySounds())
  })
)(SettingsScreen)

SettingsScreen.propTypes = {
  activityProgressType: PropTypes.string.isRequired,
  activityShowTimer: PropTypes.bool.isRequired,
  activityFeedback: PropTypes.string.isRequired,
  feedbackFrequency: PropTypes.string.isRequired,
  playSounds: PropTypes.bool.isRequired,
  toggleActivityProgressType: PropTypes.func.isRequired,
  toggleActivityTimer: PropTypes.func.isRequired,
  toggleActivityFeedback: PropTypes.func.isRequired,
  changeFeedbackFrequency: PropTypes.func.isRequired,
  routinePlayType: PropTypes.string.isRequired,
  toggleRoutinePlayType: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired,
  togglePlaySounds: PropTypes.func.isRequired,
  loggedUserEmail: PropTypes.string.isRequired
}
