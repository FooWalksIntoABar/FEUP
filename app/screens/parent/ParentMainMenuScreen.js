import React from 'react'
import { StatusBar, StyleSheet } from 'react-native'
import { Container, Header, Footer, FooterTab, Button, Icon, Title, Body, Text } from 'native-base'
import { RoutinesScreen } from './RoutinesScreen'
import RewardsScreen from './RewardsScreen'
import ActivityScreen from './ActivityScreen'
import SettingsScreen from './SettingsScreen'
import ActionButton from 'react-native-action-button'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setRoutines } from '../../actions/userActions'

export class ParentMainMenuScreen extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      selectedTab: 'activity',
      title: 'Atividade'
    }
  }

  renderSelectedTab = () => {
    switch (this.state.selectedTab) {
      case 'activity':
        return (<ActivityScreen />)
      case 'routines':
        return (<RoutinesScreen navigation={this.props.navigation} loggedUserEmail={this.props.loggedUserEmail} setRoutines={this.props.setRoutines} />)
      case 'rewards':
        return (<RewardsScreen />)
      case 'settings':
        return (<SettingsScreen navigation={this.props.navigation} loggedUserEmail={this.props.loggedUserEmail} />)
    }
  }

  getCorrespondingActionButton = () => {
    if (this.state.selectedTab === 'routines') {
      return (
        <ActionButton style={styles.actionButton} buttonColor='rgba(231,76,60,1)'>
          <ActionButton.Item className='newActivity' buttonColor='#9b59b6' title='Criar atividade' onPress={() => this.props.navigation.navigate('ActivityFormScreen')}>
            <Icon name='md-list-box' style={styles.actionButtonIcon} />
          </ActionButton.Item>
          <ActionButton.Item className='newRoutine' buttonColor='#1abc9c' title='Criar rotina' onPress={() => this.props.navigation.navigate('RoutineFormScreen')}>
            <Icon name='md-calendar' style={styles.actionButtonIcon} />
          </ActionButton.Item>
        </ActionButton>
      )
    } else if (this.state.selectedTab === 'rewards') {
      return (
        <ActionButton style={styles.actionButton} buttonColor='rgba(231,76,60,1)'>
          <ActionButton.Item className='newReward' buttonColor='#9b59b6' title='Criar prémio' onPress={() => this.props.navigation.navigate('RewardFormScreen')}>
            <Icon name='md-trophy' style={styles.actionButtonIcon} />
          </ActionButton.Item>
        </ActionButton>
      )
    }
  }

  render () {
    return (
      <Container>
        <StatusBar hidden={false} />
        <Header>
          <Body>
            <Title>{this.state.title}</Title>
          </Body>
        </Header>
        {this.renderSelectedTab()}
        <Footer>
          <FooterTab>
            <Button className='activity' active={this.state.selectedTab === 'activity'}
              onPress={() => this.setState({ selectedTab: 'activity', title: 'Atividade' })} >
              <Icon name='md-filing' />
              <Text>Atividades</Text>
            </Button>
            <Button className='routines' active={this.state.selectedTab === 'routines'}
              onPress={() => this.setState({ selectedTab: 'routines', title: 'Gerir Rotinas' })} >
              <Icon name='md-apps' />
              <Text>Rotinas</Text>
            </Button>
            <Button className='rewards' active={this.state.selectedTab === 'rewards'}
              onPress={() => this.setState({ selectedTab: 'rewards', title: 'Recompensas' })} >
              <Icon name='md-trophy' />
              <Text>Prémios</Text>
            </Button>
            <Button className='settings' active={this.state.selectedTab === 'settings'}
              onPress={() => this.setState({ selectedTab: 'settings', title: 'Opções' })}>
              <Icon name='md-settings' />
              <Text>Opções</Text>
            </Button>
          </FooterTab>
        </Footer>
        {this.getCorrespondingActionButton()}
      </Container>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    loggedUserEmail: state.user.email
  }),
  dispatch => ({
    setRoutines: routines => dispatch(setRoutines(routines))
  })
)(ParentMainMenuScreen)

ParentMainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  loggedUserEmail: PropTypes.string.isRequired,
  setRoutines: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
  actionButtonIcon: {
    fontSize: 20,
    height: 22,
    color: 'white'
  },
  actionButton: {
    marginBottom: 55
  }
})
