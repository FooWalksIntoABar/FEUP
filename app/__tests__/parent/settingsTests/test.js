import React from 'react'
import { configure, shallow } from 'enzyme'
import { ListItem } from 'native-base'
import { Alert } from 'react-native'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import '../../../__mock__/xhr-mock'

import { SettingsScreen } from '../../../screens/parent/SettingsScreen'

configure({ adapter: new Adapter() })

describe('SettingsScreen snapshot', () => {
  jest.mock('Alert', () => {
    return {
      alert: jest.fn()
    }
  })

  it('renders SettingsScreen correctly', async () => {
    const wrapper = shallow(<SettingsScreen
      navigation={{ navigate: jest.fn(), getParam: jest.fn(), pop: jest.fn() }}
      activityProgressType={''}
      activityShowTimer={false}
      activityFeedback={''}
      feedbackFrequency={''}
      playSounds={false}
      toggleActivityProgressType={jest.fn()}
      toggleActivityTimer={jest.fn()}
      toggleActivityFeedback={jest.fn()}
      changeFeedbackFrequency={jest.fn()}
      routinePlayType={''}
      toggleRoutinePlayType={jest.fn()}
      togglePlaySounds={jest.fn()}
      loggedUserEmail={''} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(ListItem).at(0).props().onPress()
    wrapper.find(ListItem).at(1).props().onPress()
    wrapper.find(ListItem).at(8).props().onPress()
    wrapper.find(ListItem).at(9).props().onPress()
    wrapper.instance().logout()
    expect(Alert.alert).toHaveBeenCalled()
    Alert.alert.mock.calls[0][2][1].onPress()
    wrapper.unmount()
  })
})
