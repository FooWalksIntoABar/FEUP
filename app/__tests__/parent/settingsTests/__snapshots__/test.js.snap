// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen snapshot renders SettingsScreen correctly 1`] = `
<Styled(Content)>
  <Styled(List)>
    <Styled(Separator)
      bordered={true}
    >
      <Styled(Text)>
        CRIANÇAS
      </Styled(Text)>
    </Styled(Separator)>
    <Styled(ListItem)
      button={true}
      icon={true}
      onPress={[Function]}
    >
      <Styled(Body)>
        <Styled(Text)>
          Adicionar nova criança
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(ListItem)
      button={true}
      icon={true}
      onPress={[Function]}
    >
      <Styled(Body)>
        <Styled(Text)>
          Remover criança
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(Separator)
      bordered={true}
    >
      <Styled(Text)>
        DEFINIÇÕES
      </Styled(Text)>
    </Styled(Separator)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Visual de progresso
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(PickerNB)
          onValueChange={[MockFunction]}
          selectedValue=""
          style={
            Object {
              "width": 125,
            }
          }
        >
          <Item
            label="Relógio"
            value="clock"
          />
          <Item
            label="Barra horizontal"
            value="bar"
          />
        </Styled(PickerNB)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Mostrar tempo restante
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Switch
          disabled={false}
          onValueChange={[MockFunction]}
          value={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Reproduzir sons
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Switch
          disabled={false}
          onValueChange={[MockFunction]}
          value={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Modo de reprodução de rotina
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(PickerNB)
          onValueChange={[MockFunction]}
          selectedValue=""
          style={
            Object {
              "width": 125,
            }
          }
        >
          <Item
            label="Escolha"
            value="choose"
          />
          <Item
            label="Automático"
            value="auto"
          />
        </Styled(PickerNB)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Tipo de feedback
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(PickerNB)
          onValueChange={[MockFunction]}
          selectedValue=""
          style={
            Object {
              "width": 125,
            }
          }
        >
          <Item
            label="Vibração"
            value="vibration"
          />
          <Item
            label="Visual"
            value="visual"
          />
          <Item
            enabled={false}
            label="Sonoro"
            value="sound"
          />
        </Styled(PickerNB)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      icon={true}
    >
      <Styled(Body)>
        <Styled(Text)>
          Intensidade de feedback
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(PickerNB)
          onValueChange={[MockFunction]}
          selectedValue=""
          style={
            Object {
              "width": 125,
            }
          }
        >
          <Item
            label="Leve"
            value="slow"
          />
          <Item
            label="Moderado"
            value="normal"
          />
          <Item
            label="Intenso"
            value="fast"
          />
        </Styled(PickerNB)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(Separator)
      bordered={true}
    >
      <Styled(Text)>
        CONTA
      </Styled(Text)>
    </Styled(Separator)>
    <Styled(ListItem)
      button={true}
      icon={true}
      onPress={[Function]}
    >
      <Styled(Body)>
        <Styled(Text)>
          Log out
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
  </Styled(List)>
</Styled(Content)>
`;
